frontend/src/metabase/dashboard/hooks/use-location-sync.ts

```diff
@@ -5,41 +5,53 @@ import { usePrevious } from "react-use";
 import { omit } from "underscore";
 
 import { DEFAULT_EMBED_DISPLAY_OPTIONS } from "metabase/dashboard/constants";
-import type { DashboardUrlHashOptions } from "metabase/dashboard/types";
 import { parseHashOptions, stringifyHashOptions } from "metabase/lib/browser";
 import { useDispatch } from "metabase/lib/redux";
 import { isNullOrUndefined } from "metabase/lib/types";
 
-const DEFAULT_DASHBOARD_EMBED_DISPLAY_OPTIONS: DashboardUrlHashOptions = {
-  ...DEFAULT_EMBED_DISPLAY_OPTIONS,
+type SYNCED_KEY = "refresh" | "fullscreen" | "theme";
+
+const DEFAULT_DASHBOARD_EMBED_DISPLAY_OPTIONS = {
+  theme: DEFAULT_EMBED_DISPLAY_OPTIONS.theme,
   fullscreen: false,
   refresh: null,
-};
+} as const;
 
-// need to type the return value as `any` to satisfy useLocationSync for now
-const getDefaultDisplayOption = (key: keyof DashboardUrlHashOptions): any =>
-  DEFAULT_DASHBOARD_EMBED_DISPLAY_OPTIONS[key];
+const getDefaultDisplayOption = <
+  Value extends typeof DEFAULT_DASHBOARD_EMBED_DISPLAY_OPTIONS[Key],
+  Key extends SYNCED_KEY,
+>(
+  key: Key,
+): Value => DEFAULT_DASHBOARD_EMBED_DISPLAY_OPTIONS[key] as Value;
 
-const isEmptyOrDefault = (value: any, key: keyof DashboardUrlHashOptions) =>
-  isNullOrUndefined(value) || value === getDefaultDisplayOption(key);
+const isEmptyOrDefault = <
+  Value extends typeof DEFAULT_DASHBOARD_EMBED_DISPLAY_OPTIONS[Key],
+  Key extends SYNCED_KEY,
+>(
+  value: Value,
+  key: Key,
+) => isNullOrUndefined(value) || value === getDefaultDisplayOption(key);
 
-export const useLocationSync = <T = any>({
+export const useLocationSync = <
+  Value extends typeof DEFAULT_DASHBOARD_EMBED_DISPLAY_OPTIONS[Key],
+  Key extends SYNCED_KEY = any,
```

Comment:
This defaults to `any` because we are already using this hook somewhere else and we don't want to have to pass more types than needed.

https://github.com/metabase/metabase/blob/19cbeacb939ed9d6953b9de804f89ff041b5bdf1/frontend/src/metabase/dashboard/hooks/use-dashboard-url-params.ts#L42-L47